subsys/bluetooth/shell/gatt.c

```diff
@@ -946,43 +949,73 @@ static int cmd_unregister_test_svc(const struct shell *sh,
 	return 0;
 }
 
-static void notify_cb(struct bt_conn *conn, void *user_data)
+static uint8_t found_attr(const struct bt_gatt_attr *attr, uint16_t handle,
+			  void *user_data)
 {
-	const struct shell *sh = user_data;
+	const struct bt_gatt_attr **found = user_data;
 
-	shell_print(sh, "Nofication sent to conn %p", conn);
+	*found = attr;
+
+	return BT_GATT_ITER_STOP;
+}
+
+static const struct bt_gatt_attr *find_attr(uint16_t handle)
+{
+	const struct bt_gatt_attr *attr = NULL;
+
+	bt_gatt_foreach_attr(handle, handle, found_attr, &attr);
+
+	return attr;
 }
 
 static int cmd_notify(const struct shell *sh, size_t argc, char *argv[])
 {
-	struct bt_gatt_notify_params params;
-	uint8_t data = 0;
+	const struct bt_gatt_attr *attr;
+	int err;
+	size_t data_len;
+	unsigned long handle;
 
-	if (!echo_enabled) {
-		shell_error(sh, "No clients have enabled notifications for the vnd1_echo CCC.");
-		return -ENOEXEC;
-	}
+	char *arg_handle = argv[1];
+	char *arg_data = argv[2];
 
-	if (argc > 1) {
-		data = strtoul(argv[1], NULL, 16);
+	err = 0;
+	handle = shell_strtoul(arg_handle, 16, &err);
+	if (err) {
+		shell_error(sh, "Handle: Not a hex number '%s'.", arg_handle);
+		return -EINVAL;
+	}
+	if (!IN_RANGE(handle, 1, 0xffff)) {
+		shell_error(sh, "Invalid handle: 0x%lx. Valid handles are [1, 0xffff].", handle);
+		return -EINVAL;
 	}
 
-	memset(&params, 0, sizeof(params));
+	attr = find_attr(handle);
+	if (!attr) {
+		shell_error(sh, "Attribute not found for handle 0x%lx on local server.", handle);
+		return -EINVAL;
+	}
 
-	params.uuid = &vnd1_echo_uuid.uuid;
-	params.attr = vnd1_attrs;
-	params.data = &data;
-	params.len = sizeof(data);
-	params.func = notify_cb;
-	params.user_data = (void *)sh;
-	SET_CHAN_OPT_ANY(params);
+	data_len = strnlen(arg_data, UINT16_MAX + 1);
+	if (data_len > UINT16_MAX) {
+		shell_error(sh, "The argument is too big to fit in a notification.");
+		return -EINVAL;
+	}
```

Comment:
Should we consider using `hex2bin` here instead? Or is your intention to only support string values?